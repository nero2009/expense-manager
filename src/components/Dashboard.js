import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom"
import {connect} from 'react-redux'
import moment from 'moment'
import numeral from 'numeral'


import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expense-total'
import {startLogout} from '../actions/auth'
// react plugin for creating charts

// @material-ui/core
import {withStyles,MuiThemeProvider, createMuiTheme}  from "@material-ui/core/styles";
import Icon from "@material-ui/core/Icon";

// @material-ui/icons
import Store from "@material-ui/icons/Store";
import DateRange from "@material-ui/icons/DateRange";
import Update from "@material-ui/icons/Update";
import Input from '@material-ui/core/Input';
import Tooltip from '@material-ui/core/Tooltip';

import Button from '@material-ui/core/Button';


// core components
import GridItem from "./components/Grid/GridItem.js";
import GridContainer from "./components/Grid/GridContainer.js";
import Table from "./components/Table/Table.js";

import Danger from "./components/Typography/Danger.js";
import Card from "./components/Card/Card.js";
import CardHeader from "./components/Card/CardHeader.js";
import CardIcon from "./components/Card/CardIcon.js";
import CardBody from "./components/Card/CardBody.js";
import CardFooter from "./components/Card/CardFooter.js";

import dashboardStyle from "./assets/jss/views/dashboardStyle.js";

const theme = createMuiTheme({
  Input:{
    border:0,
    height: "30px"
  }
})

class Dashboard extends React.Component {
  state = {
    value: 0
  };
  handleChange = (event, value) => {
    this.setState({ value });
  };

  handleChangeIndex = index => {
    this.setState({ value: index });
  };

  handleClick = () =>{
    this.props.history.push('/create');
  }

  editExpense =(id)=>{
    e.preventDefault()
    this.props.history.push(`/edit/${id}`)
  }
  
  render() {
    
    const { classes } = this.props;
   
    return (
      <div style={{padding:"10px"}}>
        <div className="dashboard-header">
            <div className="dashboard-header-title">
            <Link to="/" className="dashboard-home-link" style={{position:"absolute", left:"10px", top:"10px", textDecoration:"none", color:""}}>Expense Manager</Link>
            <MuiThemeProvider theme={theme}>
              <Input
                id="search"
                placeholder="Search"
                type="search"
                style={{ border:"none", width:"15%", fontSize:"15px"}}
              />
            </MuiThemeProvider>
            <Tooltip title="logout" style={{fontSize:"7"}}>
              <Button onClick={this.props.Logout} style={{position:"absolute", right:"0"}}>
                <Icon >settings_power</Icon>
              </Button>
            </Tooltip>
            </div>
              <Button variant="contained" color="secondary" onClick={this.handleClick} style={{position:"absolute", right:"40px", top:"50%"}}>
                Add Expense
              </Button>
            <img className="gravatar" src="/images/new_logo.png" alt=""/>
        </div>
        <GridContainer justify = "center">
          <GridItem xs={12} sm={6} md={3}>
            <Card>
              <CardHeader color="warning" stats icon>
                <CardIcon color="warning">
                  <Icon>account_balance_wallet</Icon>
                </CardIcon>
                <p className={` materialUi ${classes.cardCategory} `}>Total Expenses</p>
                <h3 className={` materialUi ${classes.cardTitle}`}>
                {`${numeral((this.props.expenseTotal)/100).format('$0,0.00')}`} 
                </h3>
              </CardHeader>
              <CardFooter stats>
                <div className={classes.stats}>
                  <Update />
                  Just Updated
                </div>
              </CardFooter>
            </Card>
          </GridItem>
          <GridItem xs={12} sm={6} md={3}>
            <Card>
              <CardHeader color="success" stats icon>
                <CardIcon color="success">
                  <Store />
                </CardIcon>
                <p className={` materialUi ${classes.cardCategory} `}>Expenses</p>
                <h3 className={` materialUi ${classes.cardTitle}`}>{`${numeral((this.props.todayExpense)/100).format('$0,0.00')}`} </h3>
              </CardHeader>
              <CardFooter stats>
                <div className={classes.stats}>
                  <DateRange />
                  Last 24 Hours
                </div>
              </CardFooter>
            </Card>
          </GridItem>
          <GridItem xs={12} sm={6} md={3}>
            <Card>
              <CardHeader color="danger" stats icon>
                <CardIcon color="danger">
                  <Icon>info_outline</Icon>
                </CardIcon>
                <p className={` materialUi ${classes.cardCategory} `}>Number of Expenses</p>
                <h3 className={` materialUi ${classes.cardCategory} `}>{this.props.expenseCount}</h3>
              </CardHeader>
              <CardFooter stats>
                <div className={classes.stats}>
                  <Update />
                  Just Updated
                </div>
              </CardFooter>
            </Card>
          </GridItem>
          
          <GridItem xs={9} sm={9} md={9}>
            <Card>
              <CardHeader color="warning">
                <h4 className={`materialUi ${classes.cardTitleWhite}`}>Expenses</h4>
                
              </CardHeader>
              <CardBody>
                <Table
                  tableHeaderColor="warning"
                  tableHead={["ID", "Description", "Created At", "Amount"," "]}
                  tableData={this.props.expenses}
                />
              </CardBody>
            </Card>
          </GridItem>
        </GridContainer>
       
      </div>
    );
  }
}
const mapStateToProps = (state)=>{
  const visibleExpenses = selectExpenses(state.expenses, state.filters)

  return{
      expenses: selectExpenses(state.expenses, { text: '',sortBy: 'date',startDate: moment().startOf('year'),endDate: moment() }),
      todayExpense: selectExpensesTotal(visibleExpenses),
      expenseCount: state.expenses.length,
      expenseTotal:selectExpensesTotal(state.expenses)
  }
}

const mapDispatchToProps =(dispatch)=>({
  Logout:()=> dispatch(startLogout())
})


Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default connect(mapStateToProps,mapDispatchToProps)(withStyles(dashboardStyle)(Dashboard));
