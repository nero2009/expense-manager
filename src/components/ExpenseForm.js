import React, { Component } from 'react'
import moment from 'moment'
import {SingleDatePicker} from 'react-dates'

import 'react-dates/initialize'

const now =moment()


class ExpenseForm extends Component {
    constructor(props){
        super(props)
        this.state ={
            description: props.expense ? props.expense.description:'',
            note: props.expense ? props.expense.note :'',
            amount: props.expense ? (props.expense.amount/100).toString():'',
            createdAt: props.expense ? moment(props.expense.createdAt): moment(),
            calendarFocused:false,
            error:'',
            amountPlaceholder:'Amount',
            desPlaceholder:'Description'
        }
    }
    

    onDescriptionChange =(e)=>{
        const description = e.target.value
        this.setState(()=>({description}))
    }

    onNoteChange =(e)=>{
        const note = e.target.value
        this.setState(()=>({note}))
    }

    onAmountChange=(e)=>{
        const amount = e.target.value
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({amount}))
        }
    }

    onDateChange =(createdAt)=>{
        if(createdAt){
            this.setState(()=>({createdAt}))
        }
        
    }

    onFocusChange =({focused})=>{
        this.setState(()=>({calendarFocused:focused}))
    }

    onFocus =(e)=>{
        const amount = this.state
        this.setState(()=>({amountPlaceholder:''}))
        if(!amount){
            this.setState(()=>({amountPlaceholder:'Amount'}))
        }
        
    }

    onFocus1 =(e)=>{
        const description = this.state
        this.setState(()=>({desPlaceholder:''}))
        if(!description){
            this.setState(()=>({amountPlaceholder:'Description'}))
        }
        
    }

    onSubmit = (e) => {
        e.preventDefault();
    
        if (!this.state.description || !this.state.amount) {
          this.setState(() => ({ error: 'Please provide description and amount.' }));
        } else {
          this.setState(() => ({ error: '' }));
          this.props.onSubmit({
            description: this.state.description,
            amount: parseFloat(this.state.amount, 10) * 100,
            createdAt: this.state.createdAt.valueOf(),
            note: this.state.note
          });
        }
      };

    render () {
        return (
            
                <form className="form" onSubmit={this.onSubmit}>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}

                    <div className="col-3 input-effect">
                        <input className="effect-21" type="text" placeholder={this.state.desPlaceholder} value={this.state.description} onChange={this.onDescriptionChange} onFocusChange={this.onFocus}/>
                        <label>Description</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <div className="col-3 input-effect">
                        <input className="effect-21" type="text" placeholder={this.state.amountPlaceholder} value={this.state.amount} onChange={this.onAmountChange} onFocusChange={this.onFocus}/>
                        <label>Amount</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths= {1}
                        isOutsideRange={()=> false}
                    />
                    <textarea 
                        className="textarea"
                        placeholder="Add a note for your Expense(optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    </textarea>
                    <div>
                        <button className="button">Save Expense</button>
                    </div>
                    
                </form>
            
        )
    }
}

export default ExpenseForm