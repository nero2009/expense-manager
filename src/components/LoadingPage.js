import React from 'react';

const LoadingPage = () => {
    return (
        <div className="container">
            <div className="circle lg">
            <div className="circle md">
                <div className="circle sm">
                <div className="circle smlr">
                </div>
                </div>
            </div>
            </div>
            <span id="loading">Loading...</span>
        </div>
    )
}

export default LoadingPage;