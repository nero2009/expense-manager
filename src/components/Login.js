import React, { Component } from 'react'
import { Link } from "react-router-dom";
import {connect} from 'react-redux'
import {startLogin, startLoginWithGoogle} from '../actions/auth'

export function Input(props){
    return(
        <div className="Input">
				<input id={props.name} placeholder={props.placeholder} />	
				<label htmlFor={props.name}></label>
			</div>
    )
}


export class Modal extends Component {
    constructor(props){
        super(props)
        this.state={
            email:'',
            username:'',
            password:''
        }
    }

    handleChange=(e)=>{
        this.setState(()=>{
            return{
                [e.target.name]:e.target.value
            }
        })
    }

    submit=(e)=>{
        console.log('clicked')
        e.preventDefault()
        this.props.login();
    }

    render () {
        console.log(this.props)
        return (
            <div className="move-right">
            <div className="Modal ">
                <form  className="ModalForm">
					<Input name="email" type="email" placeholder="nero@expense.com" value={this.state.email} onChange={this.handleChange} />
					<Input name="password" type="password" placeholder="password" value={this.state.password} onChange={this.handleChange} />
                    <div className="button-container">
                        <button className="login">Log in<i className="fa fa-fw fa-chevron-right"></i></button>
                    </div>
                    <span className="login-option-label">or connect with</span>
                    <button className="loginBtn loginBtn--google" onClick={this.submit}>
                        Login with Google
                    </button>
                    <p className="register-text">Not registered, <Link to="/create">Register now</Link></p>
                    <p className="login-text">By signing up, you agree to the <a href="#">Terms of Use</a> and <a href="#">Privacy Policy</a></p>
				</form>
               
                </div>
            </div>
        )
    }
}

const mapDispatchToProps =(dispatch)=>({
    startLogin:()=> dispatch(startLoginWithGoogle())
})

export default connect(undefined,mapDispatchToProps)(Modal)

