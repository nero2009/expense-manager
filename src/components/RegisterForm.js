import React, { Component } from 'react'
import { Link} from 'react-router-dom'

class RegisterForm extends Component {
    constructor (props) {
        super(props)
        this.state={
            firstName:'',
            lastName:'',
            email:'',
            password:''
        }
    }

    handleChange =(e)=>{
        this.setState({
            [e.target.name]:e.target.value
        })
    }

    submit =(e)=>{
        e.preventDefault()
        const {email, password} = this.state
        this.props.createUser(email,password)

    }
    
    render () {
        
        return (
            <div>
                <form className="form" onSubmit={this.onSubmit}>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}

                    <div className="form-group input-effect">
                        <input className="effect-21 register-form" type="text" name="firstName" placeholder="FirstName" value={this.state.firstName} onChange={this.handleChange} />
                        <label>First Name</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <br/>
                    <div className="form-group input-effect">
                        <input className="effect-21 register-form" type="text" name="lastName" placeholder="LastName" value={this.state.lastName} onChange={this.handleChange} />
                        <label>Last Name</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <br/>
                    <div className="form-group input-effect">
                        <input className="effect-21 register-form" type="text" name="email" placeholder="Email" value={this.state.email} onChange={this.handleChange} />
                        <label>Email</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <br/>
                    <div className="form-group input-effect">
                        <input className="effect-21 register-form" type="password" name="password" placeholder="Password" type="password" value={this.state.password} onChange={this.handleChange} />
                        <label>Password</label>
                        <span className="focus-border">
                            <i></i>
                        </span>
                    </div>
                    <br/>
                    <div>
                        <button className="register" onClick={this.submit}>Sign up<i className="fa fa-fw fa-chevron-right"></i></button>
                    </div>
                        <p style={{textAlign:"center", color:"grey"}}>Already a member? <Link to="/" style={{ color:"grey"}}>Login</Link></p>
                    </form> 
            </div>
        )
    }
}

export default RegisterForm