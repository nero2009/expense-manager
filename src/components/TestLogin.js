import React, { Component } from 'react'
import PropTypes from 'prop-types';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import Button from '@material-ui/core/Button';

import {connect} from 'react-redux'
import {startLogin,startLoginWithGoogle,createUser} from '../actions/auth'

class TestLogin extends Component {
    constructor (props) {
        super(props)
        this.state={
            email:'',
            password:''
        }
    }

    handleChange = event =>{
        this.setState({[event.target.name]:event.target.value})
    }

    login = ()=>{
        const {email,password}= this.state
        this.props.startLogin(email,password)
    }

    create = ()=>{
        const {email,password}= this.state
        this.props.createUser(email,password)
    }
    
    render () {
        const {email,password}= this.state
        const {startLogin, startLoginWithGoogle}= this.props
        return (
            <div>
                <FormControl>
                    <InputLabel>Email</InputLabel>
                    <Input id="email" name="email" value={this.state.email} onChange={this.handleChange} />
                </FormControl>
                <FormControl>
                    <InputLabel>Password</InputLabel>
                    <Input id="password" name="password" value={this.state.password} onChange={this.handleChange}/>
                </FormControl>
                <span>
                    <Button variant="contained" onClick={this.login}>Log In</Button>
                    <Button variant="contained" onClick={this.create}>Register</Button>
                    <Button variant="contained" color="primary" onClick={startLoginWithGoogle}>Log In with Google</Button>
                </span>
                
            </div>
        )
    }
}

const mapDispatchToProps =(dispatch)=>({
    startLogin:(email,pass)=> dispatch(startLogin(email,pass)),
    startLoginWithGoogle:()=> dispatch(startLoginWithGoogle()),
    createUser:(email,pass)=> dispatch(createUser(email,pass))
    
})

export default connect(undefined,mapDispatchToProps)(TestLogin)