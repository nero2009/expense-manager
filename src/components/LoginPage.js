import React from 'react';
import {connect} from 'react-redux'
import {startLogin, startLoginWithGoogle} from '../actions/auth'
import {Modal} from './Login'
import Typist from '../components/Typist/Typist'

export const LoginPage = ({startLogin}) => {
    return (
        <div className="box-layout-main">
            <div className="box-layout">
                <div className="box-layout-title">
                    <Typist ></Typist>
                </div>
                <div className="box-layout-login">
                    <Modal login={startLogin}/>
                </div>
                
            </div>
           
        </div>
    );
};

const mapDispatchToProps =(dispatch)=>({
    startLogin:()=> dispatch(startLoginWithGoogle())
})

export default connect(undefined,mapDispatchToProps)(LoginPage)