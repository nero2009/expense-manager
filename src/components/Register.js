import React from 'react'
import RegisterForm from './RegisterForm'

const Register = (props) => {
    return (
       <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Register</h1>
            </div>
        </div>
            <div className="content-container">
            <RegisterForm></RegisterForm>
        </div>
       </div>
    )
}

export default Register