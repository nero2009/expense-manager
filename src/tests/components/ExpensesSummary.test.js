import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'

test('should correctly render ExpensesSummary with 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={224} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSummary with multiple expenses',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={34} expenseTotal={255524} />)
    expect(wrapper).toMatchSnapshot()
})