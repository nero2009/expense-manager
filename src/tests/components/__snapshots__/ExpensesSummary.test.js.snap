// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.24
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    34
     
    expenses
     totalling 
    $2,555.24
  </h1>
</div>
`;
